### DISCUSSION TITLE
Wireguard Manual Config failing to save (permissions)?

### DISCUSSION THREAD

**Discussion by zubenubi** [RaspAP/raspap-webgui#1107] [General]
Hi,
I went though the manual configuration for Raspap & included wireguard as an option.
I am now trying to configure Wireguard with no joy.

The error in the UI is: 'WireGuard configuration failed to be updated'

It appears that the /etc/wireguard folder is owned by root:
drwx------  2 root     root      4096 Jan 18 18:09 wireguard

Did I miss something in the manual setup instructions ?

Thanks In Advance

**Comment by billz** [comment]
This is nearly always the result of an outdated `raspap.sudoers`. Look for this in the manual config steps and compare what's on your device to the latest release.

The wg folder is normally owned by root.

**Comment by zubenubi** [comment]
Thanks for the hint.

The step in the manual config was undertaken and the sudoers file is where it is supposed to be in /etc/sudoers.d

This is a clip from the daemon log when I try to manually configure wireguard:

17:24:09 rpi systemd[1]: Started Session 71 of user pi.
Jan 19 17:30:06 rpi lighttpd[4021]: cat: /etc/wireguard/wg0.conf: No such file or directory
Jan 19 17:30:06 rpi lighttpd[4024]: cat: /etc/wireguard/wg-server-public.key: No such file or directory
Jan 19 17:30:06 rpi lighttpd[4027]: cat: /etc/wireguard/wg-peer-public.key: No such file or directory
Jan 19 17:30:06 rpi lighttpd[4030]: cat: /etc/wireguard/client.conf: No such file or directory
Jan 19 17:30:06 rpi lighttpd[4038]: chmod: cannot access '/tmp/wireguard.log': No such file or directory
Jan 19 17:30:15 rpi lighttpd[4066]: cat: /etc/wireguard/wg0.conf: No such file or directory
Jan 19 17:30:15 rpi lighttpd[4069]: cat: /etc/wireguard/wg-server-public.key: No such file or directory
Jan 19 17:30:15 rpi lighttpd[4072]: cat: /etc/wireguard/wg-peer-public.key: No such file or directory
Jan 19 17:30:15 rpi lighttpd[4075]: cat: /etc/wireguard/client.conf: No such file or directory
Jan 19 17:30:28 rpi lighttpd[4125]: rm: cannot remove '/etc/wireguard/wg0.conf': No such file or directory
Jan 19 17:30:28 rpi lighttpd[4128]: rm: cannot remove '/etc/wireguard/wg-peer-private.key': No such file or directory
Jan 19 17:30:28 rpi lighttpd[4131]: rm: cannot remove '/etc/wireguard/wg-peer-public.key': No such file or directory
Jan 19 17:30:28 rpi lighttpd[4134]: rm: cannot remove '/etc/wireguard/client.conf': No such file or directory
Jan 19 17:30:28 rpi lighttpd[4140]: cat: /etc/wireguard/wg0.conf: No such file or directory
Jan 19 17:30:28 rpi lighttpd[4143]: cat: /etc/wireguard/wg-server-public.key: No such file or directory
Jan 19 17:30:28 rpi lighttpd[4146]: cat: /etc/wireguard/wg-peer-public.key: No such file or directory
Jan 19 17:30:28 rpi lighttpd[4149]: cat: /etc/wireguard/client.conf: No such file or directory
Jan 19 17:32:51 rpi systemd[1]: Reloading.

What might I be missing?

[snip]

**Reply by billz** [reply]
Did you install wireguard with apt?

**Reply by zubenubi** [reply]
Yes, as outlined in the manual install instructions:

`sudo apt-get install wireguard
sudo sed -i "s/\('RASPI_WIREGUARD_ENABLED', \)false/\1true/g" /var/www/html/includes/config.php
sudo systemctl enable wg-quick@wg`

Details from systemctl
`pi@rpi:~ $ sudo systemctl status wg-quick@wg
● wg-quick@wg.service - WireGuard via wg-quick(8) for wg
     Loaded: loaded (/lib/systemd/system/wg-quick@.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2022-01-20 16:02:51 EST; 54s ago
       Docs: man:wg-quick(8)
             man:wg(8)
             https://www.wireguard.com/
             https://www.wireguard.com/quickstart/
             https://git.zx2c4.com/wireguard-tools/about/src/man/wg-quick.8
             https://git.zx2c4.com/wireguard-tools/about/src/man/wg.8
    Process: 7608 ExecStart=/usr/bin/wg-quick up wg (code=exited, status=1/FAILURE)
   Main PID: 7608 (code=exited, status=1/FAILURE)
        CPU: 21ms

Jan 20 16:02:51 rpi systemd[1]: Starting WireGuard via wg-quick(8) for wg...
Jan 20 16:02:51 rpi wg-quick[7608]: wg-quick: `/etc/wireguard/wg.conf' does not exist
Jan 20 16:02:51 rpi systemd[1]: wg-quick@wg.service: Main process exited, code=exited, status=1/FAILURE
Jan 20 16:02:51 rpi systemd[1]: wg-quick@wg.service: Failed with result 'exit-code'.
Jan 20 16:02:51 rpi systemd[1]: Failed to start WireGuard via wg-quick(8) for wg.
pi@rpi :~ $
`

**Reply by billz** [reply]
Try using the installer and compare the result. I can't advise you further on debugging your manual setup, sorry

**Reply by zubenubi** [reply]
OK, so I wiped and reinstalled using a freshly formatted sd card. In the quick installer I opted to install wireguard as well.

Im still unable to manually define a wireguard server. I see the same error in the log

Is there anything I can do to get this working?

thanks

**Reply by billz** [reply]
I don't like saying "it works for me" but... unable to reproduce this. No issues creating a manual wg config (RPi OS 32-bit Lite / RPi 4b).
You'll need to debug on your own hardware / OS. Good luck.

**Reply by zubenubi** [reply]
So I now have an rpi 4b running the latest rpi os 32 bit lite. I went through the quick setup script today (twice). The first time I only installed Wireguard as an option. Thinking that it might be dependent on OpenVPN components, I did it again (from scratch and included openvpn as an option as well.

Neither seemed to change the outcome though. 

When I opt for the manual configuration of Wireguard, I click the magic button to generate the key and then press save. The message is as before ('WireGuard configuration failed to be updated'). How can I know what the save button is attempting to do (in order to debug)? Should I attempt to open the /etc/wireguard folder to all users ? Is the script that the save button runs executing as the 'www-data' user or some other user?

Thanks

**Reply by billz** [reply]
Try again with the latest release.

**Reply by zubenubi** [reply]
I re-installed on a fresh rpi3b and it now seems to work. I have some learning to do with wireguard. Thanks for the fix
